src/pages/ViewOffer.tsx

```diff
@@ -40,61 +60,50 @@ export function ViewOffer() {
       .catch(addError);
   };
 
-  const take = () => {
-    commands
-      .importOffer({ offer: offer! })
-      .then(() =>
-        commands
-          .takeOffer({
-            offer: offer!,
-            fee: toMojos(fee || '0', walletState.sync.unit.decimals),
-          })
-          .then((result) => setResponse(result))
-          .catch(addError),
-      )
-      .catch(addError);
+  const take = async () => {
+    try {
+      const result = await commands.takeOffer({
+        offer: offer!,
+        fee: toMojos(fee || '0', walletState.sync.unit.decimals),
+      });
+      setResponse(result);
+    } catch (error) {
+      addError(error);
+    }
   };
 
   return (
     <>
       <Header title='View Offer' />
 
       <Container>
-        {summary && (
-          <OfferCard summary={summary}>
-            <div className='flex flex-col space-y-1.5'>
-              <Label htmlFor='fee'>Network Fee</Label>
-              <Input
-                id='fee'
-                type='text'
-                placeholder='0.00'
-                className='pr-12'
-                value={fee}
-                onChange={(e) => setFee(e.target.value)}
-                onKeyDown={(event) => {
-                  if (event.key === 'Enter') {
-                    event.preventDefault();
-                    take();
-                  }
-                }}
-              />
+        {isLoading ? (
+          <Loading className="my-8" text={loadingStatus} />
+        ) : summary && (
+          <>
+            <OfferCard summary={summary}>
+              <div className='flex flex-col space-y-1.5'>
+                <Label htmlFor='fee'>Network Fee</Label>
+                <Input
+                  id='fee'
+                  type='text'
+                  placeholder='0.00'
+                  className='pr-12'
+                  value={fee}
+                  onChange={(e) => setFee(e.target.value)}
```

Comment:
Is there a reason to remove the onKeyDown handler?

Reply:
added this back